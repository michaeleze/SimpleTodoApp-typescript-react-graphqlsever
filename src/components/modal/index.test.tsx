import React from 'react';
import { cleanup, render, RenderResult } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Modal from './index';
import { act, Simulate } from 'react-dom/test-utils';

describe('Modal component', () => {
  let container: RenderResult<typeof import("@testing-library/dom/types/queries")>;
  const handleCloseModal = jest.fn();
  const handleUpdateTask = jest.fn();
  const handleOnChange = jest.fn();

  beforeEach(() => {
    container = render(
      <Modal
        handleCloseModal={handleCloseModal}
        handleUpdateTask={handleUpdateTask}
        modalItem={{ text: 'hello' }}
        isOpen={true}
      />
    );
  });

  afterEach(cleanup);

  it('should renders snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('should renders modal component', () => {
    const { getByText } = container;

    expect(getByText('hello')).toBeInTheDocument();
  });

  it('should renders input', () => {
    const { getByTestId } = container;

    expect(getByTestId("modal-input")).toHaveValue('hello');
  });

  it('should renders accept button', () => {
    const { getByTestId } = container;

    expect(getByTestId("button-accept")).toBeInTheDocument();
  });

  xit('should handle onChange', () => {
    const { getByTestId } = container;

    act(() => {
      Simulate.change(getByTestId("modal-input"), { target: { value: Event } });
  })

  expect(handleOnChange).toHaveBeenCalled();
});

});
