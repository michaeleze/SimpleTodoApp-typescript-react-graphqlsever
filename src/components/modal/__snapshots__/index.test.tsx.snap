// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should renders snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="ReactModal__Body--open"
  >
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div />
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div
      class="ReactModalPortal"
    >
      <div
        class="ReactModal__Overlay"
        style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(255, 255, 255, 0.75);"
      >
        <div
          aria-modal="true"
          class="ReactModal__Content"
          role="dialog"
          style="position: absolute; top: 40px; left: 40px; right: 40px; bottom: 40px; border: 1px solid #ccc; background: rgb(255, 255, 255); overflow: auto; border-radius: 4px; outline: none; padding: 20px;"
          tabindex="-1"
        >
          <table
            class="modal-content"
          >
            <tbody>
              <tr>
                <td
                  class="modal-text"
                >
                  <textarea
                    class="modal-input--field"
                    data-testid="modal-input"
                  />
                </td>
                <td>
                  <div
                    class="modal-options"
                  >
                    <button
                      class="button-accept"
                      data-testid="button-accept"
                    >
                      Accept
                    </button>
                    <button
                      class="button-reject"
                      data-testid="button-reject"
                    >
                       Cancel 
                    </button>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
